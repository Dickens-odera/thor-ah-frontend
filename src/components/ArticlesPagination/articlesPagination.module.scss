$red: #C40018;

.pagination  {
  margin: auto;
  text-align: center;
  padding: 10px;
  margin: 15px 0;
  width: 100%;

  &::after {
    content: '';
    display: block;
    clear: both;
  }

  .clear::after {
    content: '';
    display: block;
    clear: both;
  }

  ul {
    display: inline-block;
    background: #FBFBFB;
    list-style: none;
    padding: 0;
    margin: 0 auto;
    box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);
    border-radius: 40px;
    border: 1px solid #FBFBFB;
    padding: 20px 10px 15px 10px;

    @media only screen and (max-width: 600px) {
      padding: 10px 10px 5px 10px;
    }

    li {
      & {
        width: 30px;
        height: 30px;
        margin: 0 10px;
        border-radius: 50%;
        border: 1px solid rgba(196, 0, 24, 0.2);
        box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);
        display: inline-block;
        position: relative;
        font-weight: bold;
        color: $red;
        cursor: pointer;
        transition: .5s ease-in-out;

        &.active, &:hover {
          background: rgba(196, 0, 24, 0.9);
          color: #fff;
          width: 32px;
          height: 32px;
        }

        span {
          position: absolute;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -50%);
        }
      }
      
      &.dir {
        width: 25px;
        height: 25px;

        span {
          position: absolute;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -40%);
        }
      }
    }
  }
}