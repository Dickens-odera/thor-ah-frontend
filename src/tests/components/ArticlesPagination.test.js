import React from "react";
import { create } from 'react-test-renderer';
import ArticlesPagination from "../../components/ArticlesPagination/ArticlesPagination";

describe('Articles Pagination', () => {
  it("create a snapshot", () => {
    const c = create(<ArticlesPagination />);
    expect(c.toJSON()).toMatchSnapshot();
  });
})



