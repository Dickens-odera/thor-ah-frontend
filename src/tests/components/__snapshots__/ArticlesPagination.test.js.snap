// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Pagination create a snapshot 1`] = `
<div
  className="pagination"
>
  <ul
    className="clear"
  >
    <li
      className="dir"
    >
      <span>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </span>
    </li>
    <li>
      <span>
        1
      </span>
    </li>
    <li
      className="active"
    >
      <span>
        2
      </span>
    </li>
    <li>
      <span>
        3
      </span>
    </li>
    <li
      className="dir"
    >
      <span>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </span>
    </li>
  </ul>
</div>
`;

exports[`create a snapshot 1`] = `
<div
  className="pagination"
>
  <ul
    className="clear"
  >
    <li
      className="dir"
    >
      <span>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </span>
    </li>
    <li>
      <span>
        1
      </span>
    </li>
    <li
      className="active"
    >
      <span>
        2
      </span>
    </li>
    <li>
      <span>
        3
      </span>
    </li>
    <li
      className="dir"
    >
      <span>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </span>
    </li>
  </ul>
</div>
`;
